infra/database/type/testcontainers/src/main/java/org/apache/shardingsphere/infra/database/testcontainers/type/TcP6spyMySQLDatabaseType.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.database.testcontainers.type;
+
+import org.apache.shardingsphere.infra.database.core.type.DatabaseType;
+import org.apache.shardingsphere.infra.spi.type.typed.TypedSPILoader;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+/**
+ * Database type of MariaDB in testcontainers-java.
+ */
+public final class TcP6spyMySQLDatabaseType implements TestcontainersDatabaseType {
+    
+    @Override
+    public Collection<String> getJdbcUrlPrefixes() {
+        return Collections.singleton("jdbc:tc:p6spy:mysql:");
+    }
```

Comment:
- I don't see any testcontainers module for p6spy at https://github.com/testcontainers/testcontainers-java . Is the `org.testcontainers.containers.JdbcDatabaseContainerProvider` SPI implementation corresponding to the jdbcUrl prefix open-source?

Reply:
Thanks for pointing this out. I didn't understand the definition of the testcontainers module and thought it was a test case.  Maybe I shouldn't modify this module?

Reply:
- The testcontainers module of shardingsphere infra corresponds to several SPI implementations of `org.testcontainers.containers.JdbcDatabaseContainerProvider`. For example, https://github.com/testcontainers/testcontainers-java/blob/main/modules%2Fmysql%2Fsrc%2Fmain%2Fjava%2Forg%2Ftestcontainers%2Fcontainers%2FMySQLContainerProvider.java . 

- If there is no corresponding SPI implementation class, then there is actually no need to modify the infra testcontainers module.

Reply:
OK, thanks, I understand. I will resubmit the changes.

Reply:
> OK, thanks, I understand. I will resubmit the changes.

- Due to the ongoing controversy on the testcontainers side of the PR, I added documentation for this optional module at #33707 and https://shardingsphere.apache.org/document/current/cn/user-manual/shardingsphere-jdbc/optional-plugins/testcontainers/ .